packages/evm/src/precompiles/util/bls12_381.ts

```diff
@@ -0,0 +1,344 @@
+import {
+  bytesToBigInt,
+  bytesToUnprefixedHex,
+  concatBytes,
+  equalsBytes,
+  padToEven,
+  unprefixedHexToBytes,
+} from '@ethereumjs/util'
+
+import { ERROR, EvmError } from '../../exceptions.js'
+
+// base field modulus as described in the EIP
+const fieldModulus = BigInt(
+  '0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab'
+)
+
+// gas discount pairs taken from EIP-2537 `Bls12381MultiExpGasDiscount` parameter
+export const gasDiscountPairs = [
+  [1, 1200],
```

Comment:
Hmm, this is also in common, what is it doing here? :thinking:

Reply:
Not 100% sure but I'd honestly rather have it here than in `common` since it throws off the typing for the `common` params dictionary.  WDYT?

Reply:
That's also fine, but then it should be removed from common.

Reply:
Yeah, I thought the same yesterday when having a quick look and wondered if it might make sense to move this *out of Common*, to not have the generic Common configuration bloated too much and getting the typing unnecessarily extensive.